source/server/admin/config_dump_handler.cc

```diff
@@ -241,11 +241,11 @@ ConfigDumpHandler::addAllConfigToDump(envoy::admin::v3::ConfigDump& dump,
       Protobuf::FieldMask field_mask;
       ProtobufUtil::FieldMaskUtil::FromString(mask.value(), &field_mask);
       // We don't use trimMessage() above here since masks don't support
-      // indexing through repeated fields.
+      // indexing through repeated fields. We don't return error on failure
+      // because different callback return types will have different valid
+      // field masks.
       if (!checkFieldMaskAndTrimMessage(field_mask, *message)) {
-        return absl::optional<std::pair<Http::Code, std::string>>{std::make_pair(
-            Http::Code::BadRequest, absl::StrCat("FieldMask ", field_mask.DebugString(),
-                                                 " could not be successfully used."))};
+        continue;
```

Comment:
WDYT about just including in the output text the error message that we had before. IIUC you were taking a case that had an instructive error message before about what was wrong, and then making it just return an empty string.

Maybe collect the names of the fields that had invalid masks and return them in one error message?

We might also return an http error when none of the clusters had a useful mask. WDYT?

Reply:
I'm considering the case in the test, when someone just supplies `mask=bootstrap`. Intended behavior there is just that the bootstrap config is returned, with no error message. However, that mask *is* invalid for the majority of protos in the dump (which is why this was failing before).

I think that returning an HTTP error when no clusters had a useful mask is a good idea.

Reply:
sgtm.